src/lseq/ident.rs

```diff
@@ -228,19 +228,26 @@ impl<A: Actor> IdentGen<A> {
         );
 
         let mut rng = rand::rngs::OsRng;
-        let interval = BOUNDARY.min(upper - 1 - lower);
+        let interval = std::cmp::min(BOUNDARY, upper - lower);
```

Comment:
Rng::gen_range does not include the upper bound when it samples. No need to subtract 1 here.